plugins/inputs/logfile/tail/tail.go

```diff
@@ -182,11 +187,25 @@ func (tail *Tail) closeFile() {
 
 func (tail *Tail) reopen() error {
 	tail.closeFile()
+	numTries := 0
+	waitDuration := fileOpenRetryDuration
 	for {
 		var err error
 		tail.file, err = OpenFile(tail.Filename)
 		tail.curOffset = 0
 		if err != nil {
+			if os.IsPermission(err) {
+				// Specifically on Windows, reopening the file too quickly can result in an access denied error.
+				// Retry to allow for some leniency for when the file on the host becomes available.
+				tail.Logger.Debugf("Access denied on %s. Retried %d times so far", tail.Filename, numTries)
+				if numTries < fileOpenMaxRetries {
+					tail.Logger.Debugf("Sleeping for %v ms and retrying", waitDuration.Milliseconds())
+					time.Sleep(waitDuration)
+				}
+				numTries += 1
+				waitDuration *= 2
+				continue
+			}
```

Comment:
100ms + 200ms + 400ms + 800ms + 1600ms = 3100ms.
I think that is fine since it is only for an edge case on the Windows OS.
Otherwise I'd suggest adding checks in any long running functions/methods to `tail.Dying()` or `tail.Err()` and return early.